### PR TITLE
Add "max zoom" value into camera capabilities.

### PR DESCRIPTION
**Pull request by PStrelchenko**
Hello, guys!

I think it will be useful if you add "maxZoom" value into camera capabilities. Little use-case: you want to create pinch-zoom camera view, and you need max zoom level value for correct zoom step value calculations according to fingers travelling distance calculations.

### CODE REVIEW

**Comment by PStrelchenko** [comment]
@Diolor , thanks for the review!
I've made several changes according to your request, please check it.

File: fotoapparat/src/main/java/io/fotoapparat/routine/zoom/UpdateZoomLevelRoutine.kt [hunk 1]
```diff
@@ -17,7 +18,7 @@ internal fun Device.updateZoomLevel(
     zoomLevel.ensureInBounds()
     val cameraDevice = awaitSelectedCamera()
 
-    if (cameraDevice.getCapabilities().canZoom) {
+    if (cameraDevice.getCapabilities().zoom is Zoom.VariableZoom) {
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/routine/zoom/UpdateZoomLevelRoutine.kt]: java users will hate us for this but 🤷‍♂️

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package io.fotoapparat.parameter
+
+/**
+ * Zoom modes which camera can use.
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt]: can have*

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+ */
+sealed class Zoom {
+
+    object FixedZoom : Zoom()
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt]: missing doc e.g. : `The camera can only support one, fixed zoom level`

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+    data class VariableZoom(val minZoom: Int, val maxZoom: Int) : Zoom()
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt]: missing doc e.g. : `The camera can only support a variable zoom level between (and including) [minZoom] and [maxZoom] values`

File: fotoapparat/src/main/java/io/fotoapparat/parameter/SupportedParameters.kt [hunk 5]
```diff
@@ -56,10 +56,11 @@ internal class SupportedParameters(
     }
 
     /**
-     * @see Camera.Parameters.isZoomSupported
+     * @return [io.fotoapparat.parameter.Zoom.FixedZoom] if [Camera.Parameters.isZoomSupported] returns false,
+     * and [io.fotoapparat.parameter.Zoom.VariableZoom] with max zoom level otherwise.
      */
     val supportedZoom by lazy {
-        cameraParameters.isZoomSupported
+        if (cameraParameters.isZoomSupported) Zoom.VariableZoom(0, cameraParameters.maxZoom) else Zoom.FixedZoom
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/SupportedParameters.kt]: `0` is magic number so let's extract it into a const somewhere in this file

Comment (Diolor) [reply fotoapparat/src/main/java/io/fotoapparat/parameter/SupportedParameters.kt]: btw if the minValue is alway 0 we can only return maxZoom in the `VariableZoom` and add in the doc that levels are between `0 and max`

File: fotoapparat/src/main/java/io/fotoapparat/routine/zoom/UpdateZoomLevelRoutine.kt (hunk 1 above)
Comment (PStrelchenko) [reply fotoapparat/src/main/java/io/fotoapparat/routine/zoom/UpdateZoomLevelRoutine.kt]: Hmm, maybe will be better if I return "canZoom" parameter?...

File: fotoapparat/src/main/java/io/fotoapparat/parameter/SupportedParameters.kt (hunk 5 above)
Comment (PStrelchenko) [reply fotoapparat/src/main/java/io/fotoapparat/parameter/SupportedParameters.kt]: According to the Camera.Parameters docs, "setZoom" method can use values in the range [0, maxZoom], so I think min value always 0. 

Let's remove min value =)

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt (hunk 2 above)
Comment (PStrelchenko) [reply fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt]: Ok, updated.

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt (hunk 3 above)
Comment (PStrelchenko) [reply fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt]: Ok, added.

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt (hunk 4 above)
Comment (PStrelchenko) [reply fotoapparat/src/main/java/io/fotoapparat/parameter/Zoom.kt]: Ok, added.

File: fotoapparat/src/main/java/io/fotoapparat/parameter/SupportedParameters.kt (hunk 5 above)
Comment (Diolor) [reply fotoapparat/src/main/java/io/fotoapparat/parameter/SupportedParameters.kt]: let's just add a small note in the Variable zoom that the min value is always 0

File: fotoapparat/src/main/java/io/fotoapparat/routine/zoom/UpdateZoomLevelRoutine.kt (hunk 1 above)
Comment (Diolor) [reply fotoapparat/src/main/java/io/fotoapparat/routine/zoom/UpdateZoomLevelRoutine.kt]: Let's leave it as is. We focus on Kotlin first after all

**Comment by Diolor** [comment]
Thanks! It will be in the next release. If you need it immediately you can use a `-snapshot` FA version

**Comment by PStrelchenko** [comment]
Thanks!

**Comment by PStrelchenko** [comment]
@Diolor , hello! Can you tell me, when you planned to release the new version of Fotoapparat with this functions?..

**Pull request merged**